gossipsub_test.go

```diff
@@ -23,6 +23,26 @@ func getGossipsubs(ctx context.Context, hs []host.Host, opts ...Option) []*PubSu
 	return psubs
 }
 
+func waitForMeshConstruction(topic string, psubs []*PubSub) {
```

Comment:
Races with the heartbeat thread for GossipsubRouter.mesh[T]

Reply:
yep, if your function is very computationally inexpensive you can generally use something along the lines of this:

```
        done := make(chan resultType,1)
	p.eval <- func() {
               done <- myFn()
	}
        return <-done
```

Note: the channel closure is only going to be needed if your function needs to be synchronous. Also, you may want to add in `select` statements that check for an expired context if you want to reuse these functions a bunch.